src/main/java/net/mcreator/generator/blockly/BlocklyBlockCodeGenerator.java

```diff
@@ -1,268 +1,267 @@
-						found = true;
-
-						// check if nesting statement block that already provides any dependency with
-						// a same name, to avoid compile errors due to variable redefinitions
-						if (statementInput.provides != null) {
-							for (Dependency dependency : statementInput.provides) {
-								if (master.checkIfDepProviderInputsProvide(dependency)) {
-									master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
-											"Statement input " + statementInput.name
-													+ " provides dependencies already provided by parent statement inputs."));
-									return; // no need to do further processing, this needs to be resolved first by the user
-								}
-							}
-						}
-
-						master.pushDepProviderInputStack(statementInput);
-						String generatedCode = BlocklyToCode.directProcessStatementBlock(master, element);
-						master.popDepProviderInputStack();
-
-						dataModel.put("statement$" + statementInput.name, generatedCode);
-
-						break; // found, no need to look other elements
-					}
-				}
-				if (!found) {
-					dataModel.put("statement$" + statementInput.name, "");
-					master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.WARNING,
-							"Statement input " + statementInput.name + " on block " + type + " is empty."));
-				}
-			}
-		}
-
-		if (toolboxBlock.getRequiredAPIs() != null) {
-			for (String required_api : toolboxBlock.getRequiredAPIs()) {
-				if (!master.getWorkspaceSettings().getMCreatorDependencies().contains(required_api)) {
-					master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
-							"Block " + type + " requires " + required_api
-									+ " enabled in workspace settings, or the current generator does not support it"));
-				}
-			}
-		}
-
-		// add custom warnings if present
-		if (toolboxBlock.getWarnings() != null) {
-			for (String warning : toolboxBlock.getWarnings()) {
-				master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.WARNING,
-						L10N.t("blockly.warning." + warning, type)));
-			}
-		}
-
-		if (templateGenerator != null) {
-			dataModel.put("customBlockIndex", customBlockIndex);
-
-			if (additionalData != null) {
-				dataModel.putAll(additionalData);
-			}
-
-			String code = templateGenerator.generateFromTemplate(type + "." + templateExtension + ".ftl", dataModel);
-			master.append(code);
-		}
-
-		customBlockIndex++;
-	}
-
-	// supported blocks by type cache for performance
-	private final Map<IBlockGenerator.BlockType, String[]> blocks_machine_names = new HashMap<>();
-
-	public String[] getSupportedBlocks(IBlockGenerator.BlockType blockType) {
-		if (blocks_machine_names.containsKey(blockType)) {
-			return blocks_machine_names.get(blockType);
-		} else {
-			String[] retval = blocks.values().stream().filter(block -> block.type == blockType)
-					.map(block -> block.machine_name).toArray(String[]::new);
-			blocks_machine_names.put(blockType, retval);
-			return retval;
-		}
-	}
-}
+/*
```

Comment:
Not sure what you did, but now this whole file is marked as changed, this is not ok as it is not possible to review it

Reply:
I said about this, read my previous comment.
>Not sure what you did

@KlemenDEV I resolved merge conflicts via GitHub (how else should I do that?).

Reply:
You should resovle them with git command line too. I will use external diff to compare

Reply:
:heavy_check_mark: Fixed :upside_down_face: